docs/cpp/source/notes/tensor_cuda_stream.rst

```diff
@@ -0,0 +1,189 @@
+Tensor CUDA Stream API 
+======================
+
+A `CUDA Stream`_ is a linear sequence of execution that belongs to a specific device.
+Pytorch C++ API has defined a CUDAStream class and a few useful functions to support CUDA Stream operations.
+You can find them in `CUDAStream.h`_. This note provides more details on how to use Pytorch C++ CUDA Stream APIs.
+
+.. _CUDA Stream: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#streams
+.. _CUDAStream.h: https://pytorch.org/cppdocs/api/file_c10_cuda_CUDAStream.h.html#file-c10-cuda-cudastream-h
+.. _CUDAStreamGuard.h: https://pytorch.org/cppdocs/api/structc10_1_1cuda_1_1_c_u_d_a_stream_guard.html
+
+Get CUDA stream
+*****************
+
+Pytorch C++ API provides the following ways to get CUDA stream:
+
+1. Get a new stream from the CUDA stream pool, streams are preallocated from the pool and returned in a round-robin fashion.
+
+.. code-block:: cpp
+  
+  CUDAStream getStreamFromPool(const bool isHighPriority = false, DeviceIndex device = -1);
```

Comment:
This is not an issue for this PR, but in general it's too bad that we're inconsistent about the naming of our DeviceIndex parameters. That is, I'm sure users would expect us to use device_index here like we do in getDefaultCUDAStream.

Reply:
Yes, device and device_index are used inconsistently among cuda stream/guard/util apis, not nice, need a fix.